spec/suites/layer/marker/MarkerSpec.js

```diff
@@ -330,4 +330,18 @@ describe("Marker", function () {
 		});
 
 	});
+
+	describe('removal', function () {
+		it('should call destroyIcon', function () {
+			var marker = new L.Marker([0, 0], {icon: icon1});
+			var destroyIconSpy = sinon.spy(icon1, 'destroyIcon');
+			var destroyShadowSpy = sinon.spy(icon1, 'destroyShadow');
+			map.addLayer(marker);
+
+			map.removeLayer(marker);
+
+			expect(destroyIconSpy.calledOnce).to.equal(true);
```

Comment:
Given the issue I mentioned below, it would be nice to not only check that the methods are called, but also that they are called with the expected arguments (the icon element for `destroyIcon` and the shadow for `destroyShadow`).